python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -323,6 +328,42 @@ def write(self, value: Any):
             )
             raise TypeError(msg) from e
 
+        # serialized_value.total_bytes *only* includes the size of the data. It does not
+        # include the size of the metadata, so we must account for the size of the
+        # metadata explicitly.
+        size = serialized_value.total_bytes + len(serialized_value.metadata)
+        if size > self._typ.buffer_size_bytes:
+            # Now make the channel backing store larger.
+            self._typ.buffer_size_bytes = size
+            # TODO(jhumphri): I assume the current writer ref object is automatically
+            # freed when the reference to it is destroyed in the line below?
+            prev_writer_ref = self._writer_ref
+            self._writer_ref = _create_channel_ref(self, self._typ.buffer_size_bytes)
```

Comment:
Suggest separating out object allocation into a helper method too, so that it can be shared between this and the constructor.

Reply:
Could you please elaborate on how to do this? It's not clear to me how to decompose this beyond what we have in _create_channel_ref(). Thanks

Reply:
Ah it seemed to me that this logic [here](https://github.com/ray-project/ray/blob/08b4b49de8e1ca8466abad91ad85953f3042e224/python/ray/experimental/channel/shared_memory_channel.py#L219-L239) and [here](https://github.com/ray-project/ray/blob/08b4b49de8e1ca8466abad91ad85953f3042e224/python/ray/experimental/channel/shared_memory_channel.py#L362-L374) was basically the same. I haven't looked super carefully though, so maybe not worth factoring out.

Reply:
Done.